tools/packaging/packaging.psm1

```diff
@@ -1229,7 +1229,558 @@ function New-ZipPackage
+        <title>PowerShellRuntime</title>
+        <authors>Microsoft</authors>
+        <owners>microsoft,powershell</owners>
+        <requireLicenseAcceptance>false</requireLicenseAcceptance>
+        <description>PowerShell runtime for hosting PowerShell</description>
+        <copyright>© Microsoft Corporation.  All rights reserved.</copyright>
+        <language>en-US</language>
+        <dependencies>
+            <group targetFramework=".NETCoreApp2.0">
+            </group>
+        </dependencies>
+    </metadata>
+</package>
+'@
+
+    $nuspecTemplate = $nuspecTemplate.Replace('!id!', $PackageId)
+    $nuspecTemplate = $nuspecTemplate.Replace('!replace_me!', $PackageVersion)
+    $nuspecObj = [xml] $nuspecTemplate
+
+    if ($Dependency -ne $null) {
+
+        foreach($dep in $Dependency) {
+            # Each item is [tuple[ [tuple[string, string]], [tuple[string, string]] ]
+            $d = $nuspecObj.package.metadata.dependencies.group.AppendChild($nuspecObj.CreateElement("dependency"))
+
+            # 'id' and value
+            $d.SetAttribute($dep.Item1.Item1, $dep.Item1.Item2)
+
+            # 'version' and value
+            $d.SetAttribute($dep.Item2.Item1, $dep.Item2.Item2)
+        }
+    }
+
+    $nuspecObj.Save($filePath)
+}
+
+function New-ReferenceAssembly
+{
+    param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $Unix64BinPath,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $RefAssemblyDestinationPath,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $RefAssemblyVersion,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $GenAPIToolPath,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $SnkFilePath
+    )
+
+    $genAPIFolder = New-TempFolder
+    $smaProjectFolder = New-Item -Path "$genAPIFolder/System.Management.Automation" -ItemType Directory -Force
+    $smaCs = Join-Path $smaProjectFolder 'System.Management.Automation.cs'
+    $smaCsFiltered = Join-Path $smaProjectFolder 'System.Management.Automation_Filtered.cs'
+
+    Write-Verbose -Verbose "Working directory: $genAPIFolder."
+
+    #region GenAPI
+
+    $genAPIExe = Get-ChildItem -Path "$GenAPIToolPath/*GenAPI.exe" -Recurse
+
+    if(-not (Test-Path $genAPIExe))
+    {
+        throw "GenAPI.exe was not found at: $GenAPIToolPath"
+    }
+
+    Write-Verbose -Verbose "GenAPI nuget package saved and expanded."
+
+    $unixSMAPath = Join-Path $Unix64BinPath "System.Management.Automation.dll"
+
+    if(-not (Test-Path $unixSMAPath))
+    {
+        throw "System.Management.Automation.dll was not found at: $Unix64BinPath"
+    }
+
+    $genAPIArgs = "$unixSMAPath","-libPath:$Unix64BinPath"
+    Write-Verbose -Verbose "GenAPI cmd: $genAPIExe $genAPIArgsString"
+
+    & $genAPIExe $genAPIArgs | Out-File $smaCs -Force
```

Comment:
use `start-nativeexecution` to capture errors?

Reply:
Fixed